server/models/SiteMetadataModel/index.js

```diff
@@ -1,15 +1,21 @@
 import { collections } from '../../utils/mongoCollections'
 
+const queryOptions = {
+  upsert: true,
+  returnDocument: 'after',
+}
 const SiteMetadataModel = {
   findOne: async (db, query) =>
     await db.collection(collections.metadata).findOne(query),
-  upsert: async (db, query, metadataAttributes) =>
-    await db
-      .collection(collections.metadata)
-      .findOneAndUpdate(query, metadataAttributes, {
-        upsert: true,
-        returnDocument: 'after',
-      }),
+  upsert: async (db, query, { setAttributes, addToSetAttributes }) =>
+    await db.collection(collections.metadata).findOneAndUpdate(
+      query,
+      {
+        $set: { ...(setAttributes || {}), updatedAt: new Date() },
```

Comment:
do we need both setAttributes and addToSetAttributes? I'm thinking everything except `updatedAt` could be in the `addToSetAttributes` so this could be

```js
upsert: async (db, query, attrs) =>
  await db
    .collection(collections.metadata)
    .findOneAndUpdate({
      query,
      {
        $set: { updatedAt: new Date() },
        $addToSet: attrs,
      },
      queryOptions,
    })
```

Reply:
yes we need both, the $set operator is what updates the fields that are already present, and $addToSet appends new elements to array fields, and the array field here is called participants.